s3adapter/driver.go

```diff
@@ -237,6 +237,7 @@ func (d *S3Driver) DirContents(path string) ([]os.FileInfo) {
 
 // DeleteDir would delete a directory, but isn't currently implemented
 func (d *S3Driver) DeleteDir(path string) bool {
+        d.DeleteFile(path+"/")
```

Comment:
delete a dir just delete "dir_file", but not delete all objects under this "dir"